### ISSUE TITLE
interest on merging efforts?

### ISSUE THREAD

**Issue by v1k0d3n** [skippbox/kubeadm-centos#2]
@sebgoa would you be interested in merging efforts with a project that i've worked on for my community development team?

the repo is: https://github.com/att-comdev/halcyon-kubernetes

merging would probably be good. my thought is to merge the playbook efforts with a global_var declaration for centos/ubuntu (and other os's that become supported), and then just use the repo as a submodule for a vagrant deployment that can be multi-provider.

what are your thoughts? i could have our team help maintain both, and we can use travis or something else to maintain builds, if you're interested. merging could capture a broader audience.

**Comment by sebgoa** [comment]
@v1k0d3n why not ! though honestly this was just a quick proof of concept, we are not intending to develop this into a full production playbook. Definitely we will stick to CentOS 

Are you guys going to Kubecon, we could meet there.

**Comment by errordeveloper** [comment]
@davidkbainbridge has also shared an Ubuntu playbook with Vagrant: https://github.com/davidkbainbridge/k8s-playground.

**Comment by v1k0d3n** [comment]
@errordeveloper this is a good recommendation too. would be fine with either, but would like to be able to declare multiple nodes and deploy to cloud (which would be better outside of the roles anyway; i was going to separate this this soon). do you know the owner of that repo? i think it would be good to merge these efforts for developers who really want multi-node deployment into a local environment.

**Comment by davidkbainbridge** [comment]
Happy to merge. My plan was to migrate the
https://github.com/davidkbainbridge/k8s-playground repo so that the ansible
script was a bit more separate from the vagrant bits. This way you could
provide an inventory file and use kubeadm to get a cluster working on
physical or virtual nodes.

**Comment by v1k0d3n** [comment]
This is great news! Do you guys want to get together on a hangout to discuss merging these efforts? I'll be back in town after the Openstack Summit; so say...next Friday?

**Comment by davidkbainbridge** [comment]
Next Friday could work for me. I am in the America/Los_Angeles time zone (PDT).

**Comment by errordeveloper** [comment]
I've added CentOS support to my fork of @davidkbainbridge repo (https://github.com/errordeveloper/kubernetes-ansible-vagrant), and I managed to do so without disabling firewalld and SELinux.

**Comment by v1k0d3n** [comment]
@errordeveloper could we combine some deployment strategies with the halcyon repo and this one? this is why I wanted to schedule a discussion, but i'm currently traveling. we may have a slight opinion on this, because we have larger deployment PoC effort. i would hate to keep/maintain two repos that do the same thing (now we have three). once we combine, maybe this would be good to incubate multimode/Vagrant solution for development?

**Comment by v1k0d3n** [comment]
@davidkbainbridge / @errordeveloper 
btw, is everyone going to kubecon? maybe we can just talk about it there?

**Comment by sebgoa** [comment]
Yes, that might be easier to talk at kubecon. Fwiw, kargo which is going to be incubated, will contain kubeadm support.

**Comment by v1k0d3n** [comment]
definitely a fan of kargo getting incubated (because it's needed), and we support this. the approach to deploying kubeadm is different though, it will eventually remove some of the need for CAPS, is a bit closer to a cloud-native approach; which is the whole reason we started looking at. do we need the whole Ansible playbook model in kargo just to get a development multi-node vagrant deployment? (just a hypothetical question i've been pondering lately).

**Comment by v1k0d3n** [comment]
@davidkbainbridge @sebgoa can you email me so we can get together at Kubecon or started with merging some of these "vagrant" developmental efforts? my email is bjozsa@jinkit.com.

**Comment by davidkbainbridge** [comment]
I will send an email, but right now I am not planning to attend Kubecon. I check the site, but see it is sold out at this time. It also happens to be next week and I am in Paris this week for an ONOS event, so being home for a week isn't a bad thing. If there is a way to participate remotely I would be happy to do so.
